src/lib/storage/storage_manager.cpp

```diff
@@ -53,17 +57,57 @@ std::vector<std::string> StorageManager::table_names() const {
   return table_names;
 }
 
+void StorageManager::add_view(const std::string& name, std::shared_ptr<AbstractASTNode> view) {
+  Assert(_tables.find(name) == _tables.end(),
+         "Cannot add view " + name + " - a table with the same name already exists");
+  Assert(_views.find(name) == _views.end(), "A view with the name " + name + " already exists");
+  DebugAssert(view->subtree_is_read_only(), "Views can only hold read-only SELECTs");
+
+  _views.insert(std::make_pair(name, std::move(view)));
```

Comment:
Suggestion: `emplace` would be less typing: http://en.cppreference.com/w/cpp/container/unordered_map/emplace